subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaBasePlugin.java

```diff
@@ -336,6 +338,10 @@ private void configureBuildDependents(Project project) {
 
     private void configureTest(final Project project, final JavaPluginExtension javaPluginExtension) {
         project.getTasks().withType(Test.class).configureEach(test -> configureTestDefaults(test, project, javaPluginExtension));
+        project.getTasks().withType(Jar.class).configureEach(jar -> {
+            TaskCollection<Test> testTasks = project.getTasks().withType(Test.class);
+            testTasks.forEach(test -> jar.shouldRunAfter(test));
```

Comment:
Does this support configuration avoidance?

Reply:
I'm looking into that currently.